### PR TITLE
chore: update to go 1.21.6 in docker

### PR DESCRIPTION
**Pull request by dargudear-google**
/kind feature

update to go 1.21.5 for docker builds

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 ARG BASEIMAGE=registry.k8s.io/build-image/debian-base:bookworm-v1.0.0
 
-FROM golang:1.21@sha256:337543447173c2238c78d4851456760dcc57c1dfa8c3bcd94cbee8b0f7b32ad0 as builder
+FROM golang:1.21@sha256:2ff79bcdaff74368a9fdcb06f6599e54a71caf520fd2357a55feddd504bcaffb as builder
```
Comment (aramase) [inline docker/Dockerfile:17]: The sha for latest `1.21.5` image is [`135ba6c2ae2a9a022f53f3eee24ef213bc62e876811cd9582b91045168206873`]( https://hub.docker.com/layers/library/golang/1.21/images/sha256-135ba6c2ae2a9a022f53f3eee24ef213bc62e876811cd9582b91045168206873?context=explore)

File: docker/windows.Dockerfile [hunk 2]
```diff
@@ -17,7 +17,7 @@ ARG BASEIMAGE_CORE=gcr.io/k8s-staging-e2e-test-images/windows-servercore-cache:1
 
 FROM --platform=linux/amd64 ${BASEIMAGE_CORE} as core
 
-FROM --platform=$BUILDPLATFORM golang:1.21@sha256:337543447173c2238c78d4851456760dcc57c1dfa8c3bcd94cbee8b0f7b32ad0 as builder
+FROM --platform=$BUILDPLATFORM golang:1.21@sha256:2ff79bcdaff74368a9fdcb06f6599e54a71caf520fd2357a55feddd504bcaffb as builder
```
Comment (aramase) [inline docker/windows.Dockerfile:20]: same comment as above

File: test/e2eprovider/Dockerfile [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.21@sha256:337543447173c2238c78d4851456760dcc57c1dfa8c3bcd94cbee8b0f7b32ad0 as builder
+FROM golang:1.21@sha256:2ff79bcdaff74368a9fdcb06f6599e54a71caf520fd2357a55feddd504bcaffb as builder
```
Comment (aramase) [inline test/e2eprovider/Dockerfile:15]: same comment as above

File: .local/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.21-alpine@sha256:eeaab088668869c65a2ee0c2c4df7f8c9920140ede7fba3b777cf5b7e9fdbb69
+FROM golang:1.21-alpine@sha256:55f716237933c85cee01748700755b4ac8736fb1ca974c9aed051691b68d6dc2
```
Comment (aramase) [inline .local/Dockerfile:1]: The sha for latest `1.21.5` image is [`101168f254556b660b19fd7c542cde4cdc963053573e67825ed16b8fe20e1ff8`](https://hub.docker.com/layers/library/golang/1.21-alpine/images/sha256-101168f254556b660b19fd7c542cde4cdc963053573e67825ed16b8fe20e1ff8?context=explore)

**Review by aramase — CHANGES_REQUESTED** [review]
Thank you for the PR! Added comments, PTAL!

After the comments are solved, please rebase and squash to a single commit.

**Review by aramase — COMMENTED** [review]
`1.21.6` base images are already available, so I would suggest bumping to that instead of `1.21.5`

**Comment by dargudear-google** [comment]
> `1.21.6` base images are already available, so I would suggest bumping to that instead of `1.21.5`

Done. PTAL.

**Review by aramase — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**